File: src/main/java/axirassa/ioc/IocStandalone.java

package axirassa.ioc;

import org.apache.tapestry5.ioc.Registry;
import org.apache.tapestry5.ioc.RegistryBuilder;

public class IocStandalone {

	public static Registry registry = null;


	public static RegistryBuilder initRegistryBuilder() {
		RegistryBuilder builder = new RegistryBuilder();
		builder.add(DAOModule.class);
		builder.add(FlowsModule.class);
		builder.add(MessagingModule.class);
		builder.add(LoggingModule.class);

		return builder;
	}


	public static void init() {
		init(initRegistryBuilder());
	}


	public static void init(RegistryBuilder builder) {
		if (registry == null) {
			registry = builder.build();
			registry.performRegistryStartup();
		}
	}


	public static <T> T autobuild(Class<T> classObject) {
		if (registry == null)
			init();

		return registry.autobuild(classObject);
	}

}

File: src/main/java/axirassa/ioc/IocStandalone.scala
package axirassa.ioc

import org.apache.tapestry5.ioc.Registry
import org.apache.tapestry5.ioc.RegistryBuilder

class IocStandalone {

    var registry : Registry = _

    def initRegistryBuilder() = {
        val builder = new RegistryBuilder()
        builder.add(classOf[DAOModule])
        builder.add(classOf[FlowsModule])
        builder.add(classOf[MessagingModule])
        builder.add(classOf[LoggingModule])

        classOf[IocStandalone]

        builder
    }

    def init() {
        init(initRegistryBuilder())
    }

    def init(builder : RegistryBuilder) {
        if (registry == null) {
            registry = builder.build()
            registry.performRegistryStartup()
        }
    }

    def autobuild[T](classObject : Class[T]) = {
        if (registry == null)
            init()

        registry.autobuild(classObject)
    }
}

object IocStandalone extends IocStandalone
